install_spot_ros2.sh

```diff
@@ -20,7 +20,8 @@ else
 fi
 
 # Install ROS dependencies
-sudo apt install -y ros-$ROS_DISTRO-joint-state-publisher-gui ros-$ROS_DISTRO-tf-transformations ros-$ROS_DISTRO-xacro ros-$ROS_DISTRO-depth-image-proc
+# TODO(jschornak-bdai): use rosdep to install these packages by parsing dependencies listed in package.xml
```

Comment:
I assume this is a WIP, so I'm just going to leave a sticky note here.

Reply:
Let's address this one in a different PR -- the current version of this follows the same pattern as installing the previous dependencies, while the TODO proposes changing how dependency installation works in a deeper way.